pallets/network-score/src/tests.rs

```diff
@@ -312,3 +312,90 @@ fn test_register_rating_id_already_exists() {
 		);
 	});
 }
+
+#[test]
+fn test_revoke_rating_id_already_exists() {
+	// Define test parameters
+	let creator = DID_00.clone();
+	let author = ACCOUNT_00.clone();
+	let message_id = BoundedVec::try_from([72u8; 10].to_vec()).unwrap();
+	let entity_uid = BoundedVec::try_from([73u8; 10].to_vec()).unwrap();
+	let provider_uid = BoundedVec::try_from([74u8; 10].to_vec()).unwrap();
+	let entry = RatingInputEntryOf::<Test> {
+		entity_uid,
+		provider_uid,
+		total_encoded_rating: 250u64,
+		count_of_txn: 7u64,
+		entity_type: EntityTypeOf::Logistic,
+		rating_type: RatingTypeOf::Overall,
+		provider_did: creator.clone(),
+	};
+	let entry_digest =
+		<Test as frame_system::Config>::Hashing::hash(&[&entry.encode()[..]].concat()[..]);
+	let raw_space = [2u8; 256].to_vec();
+	let space_digest = <Test as frame_system::Config>::Hashing::hash(&raw_space.encode()[..]);
+	let space_id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_digest.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+	let space_id: SpaceIdOf = generate_space_id::<Test>(&space_id_digest);
+	let auth_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_id.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+	let authorization_id: AuthorizationIdOf =
+		Ss58Identifier::create_identifier(&auth_digest.encode()[..], IdentifierType::Authorization)
+			.unwrap();
+	let id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[
+			&entry_digest.encode()[..],
+			&entry.entity_uid.encode()[..],
+			&message_id.encode()[..],
+			&space_id.encode()[..],
+			&creator.clone().encode()[..],
+		]
+		.concat()[..],
+	);
+
+	let identifier =
+		Ss58Identifier::create_identifier(&(id_digest).encode()[..], IdentifierType::Rating)
+			.unwrap();
+	let rating_entry = RatingEntries::<Test>::get(&identifier);
+
+	// Unwrap the Option to access the inner RatingEntry struct
+	let mut rating_entry = rating_entry.unwrap();
+
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+
+		assert_ok!(Space::create(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			space_digest
+		));
+		assert_ok!(Space::approve(RawOrigin::Root.into(), space_id, 3u64));
+
+		assert_ok!(Score::register_rating(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			entry.clone(),
+			entry_digest,
+			message_id.clone(),
+			authorization_id.clone(),
+		));
+
+		// Remove message_id and provider_did from entries
+		<MessageIdentifiers<Test>>::remove(message_id.clone(), creator.clone());
+		// Entry type of should not be anything other than Debit
+		rating_entry.entry_type = EntryTypeOf::Debit;
+		RatingEntries::<Test>::insert(&identifier, &rating_entry);
+
+		assert_err!(
+			Score::revise_rating(
```

Comment:
Revise rating expects entry type to be of `Debit` in the storage. To test this can be done by 
- Calling `revoke_rating` which does exactly this ( Ideal ).
- Setting entry-type to be of `DEBIT` in storage ( Not Recommended in this case since we can do through a method from 1 ).

Reply:
Ohk @vatsa287 , Which method from 1 ?

Reply:
@zeel991 execute `revoke_rating()`